### PR TITLE
Ja/lib spring3 0 13

### PR DESCRIPTION
**Pull request by aurambaj**

### CODE REVIEW

File: webapp/recording.jfr
Comment (maallen) [inline webapp/recording.jfr:1]: Assume this was added to a commit by a mistake

File: cli/pom.xml [hunk 1]
```diff
@@ -145,7 +145,7 @@
                 <artifactId>maven-surefire-plugin</artifactId>
                 <configuration>
                     <argLine>-Dspring.config.additional-location=optional:file://${user.home}/.l10n/config/cli/
-                        -Dspring.profiles.active=${user.name},test -Xmx1024m
+                        -Dspring.profiles.active=${user.name},test -Xmx2048m
```
Comment (maallen) [inline cli/pom.xml:148]: Is the memory heap increase for the tests still required?

File: pom.xml [hunk 2]
```diff
@@ -185,7 +185,8 @@
                         <groupId>org.apache.maven.plugins</groupId>
                         <artifactId>maven-compiler-plugin</artifactId>
                         <configuration>
-                            <debug>false</debug>
+                            <!-- TODO(ja-lib) can this be causing the warning? -->
+<!--                            <debug>false</debug>-->
```
Comment (maallen) [inline pom.xml:189]: It's my understanding that the warning is coming from a dependency class that isn't compiled with `parameters` so it's falling back to debug. See https://github.com/spring-projects/spring-framework/wiki/Upgrading-to-Spring-Framework-6.x#parameter-name-retention
 
When I compile our code with `<parameters>true</parameters>` the classes that that log gets printed for are all JPARepository classes:

com.box.l10n.mojito.service.commit.CommitToPullRunRepository
com.box.l10n.mojito.service.pullrun.PullRunAssetRepository
com.box.l10n.mojito.service.pullrun.PullRunTextUnitVariantRepository
com.box.l10n.mojito.service.assetTextUnit.AssetTextUnitRepository
com.box.l10n.mojito.service.tm.TMTextUnitVariantCommentRepository
com.box.l10n.mojito.service.assetExtraction.AssetTextUnitToTMTextUnitRepository
com.box.l10n.mojito.service.branch.BranchRepository
com.box.l10n.mojito.service.assetcontent.AssetContentRepository
com.box.l10n.mojito.service.commit.CommitToPushRunRepository
com.box.l10n.mojito.service.pushrun.PushRunRepository
com.box.l10n.mojito.service.pushrun.PushRunAssetRepository
com.box.l10n.mojito.service.pushrun.PushRunAssetTmTextUnitRepository
com.box.l10n.mojito.service.security.user.UserRepository
com.box.l10n.mojito.service.security.user.AuthorityRepository
com.box.l10n.mojito.service.commit.CommitRepository
com.box.l10n.mojito.service.translationkit.TranslationKitRepository
com.box.l10n.mojito.service.translationkit.TranslationKitTextUnitRepository
com.box.l10n.mojito.service.branch.BranchStatisticRepository
com.box.l10n.mojito.service.screenshot.ScreenshotRepository
com.box.l10n.mojito.service.screenshot.ScreenshotTextUnitRepository
com.box.l10n.mojito.service.thirdparty.ThirdPartyScreenshotRepository
com.box.l10n.mojito.service.thirdparty.ThirdPartyTextUnitRepository
com.box.l10n.mojito.service.gitblame.GitBlameRepository
com.box.l10n.mojito.service.branch.BranchTextUnitStatisticRepository
com.box.l10n.mojito.service.branch.notification.BranchNotificationRepository
com.box.l10n.mojito.service.pluralform.PluralFormForLocaleRepository
com.box.l10n.mojito.service.repository.statistics.StatisticsScheduleRepository
com.box.l10n.mojito.service.thirdparty.ThirdPartyFileChecksumRepository


One fix is to add @Param("<paramName">) in front of each method parameter in these classes, I've verified this with one class.

File: restclient/src/main/java/com/box/l10n/mojito/rest/client/RepositoryClient.java [hunk 3]
```diff
@@ -52,6 +52,13 @@ public Repository getRepositoryById(Long repositoryId) {
         getBasePathForResource(repositoryId), Repository.class);
   }
 
+  // TODO(ja-lib) remove just testing
```
Comment (maallen) [inline restclient/src/main/java/com/box/l10n/mojito/rest/client/RepositoryClient.java:55]: Does this need to be removed still?

**Review by maallen — APPROVED** [review]
Approving so as to not block merging later

File: cli/pom.xml (hunk 1 above)
Comment (aurambaj) [reply cli/pom.xml]: i don't think so, I added it when there was huge hibernate query generated and it was working only with mysql anyway I think. I've removed it and run a few test and it looked good

File: restclient/src/main/java/com/box/l10n/mojito/rest/client/RepositoryClient.java (hunk 3 above)
Comment (aurambaj) [reply restclient/src/main/java/com/box/l10n/mojito/rest/client/RepositoryClient.java]: yes, thanks!

**Pull request closed without merging**